src/cuda/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+set(QUICK_CUDA_CUDA_SOURCE gpu.cu gpu_type.cu gpu_get2e.cu)
+     
+set(QUICK_CUDA_FORTRAN_SOURCE gpu_write_info.f90)
+
+set_property(SOURCE ${QUICK_CUDA_FORTRAN_SOURCE} PROPERTY COMPILE_OPTIONS ${OPT_FFLAGS})
+
+#sadly FindCUDA doesn't support OBJECT libraries, so we have to use a static cuda_add_library
+cuda_add_library(quick_cuda STATIC ${QUICK_CUDA_FORTRAN_SOURCE} ${QUICK_CUDA_CUDA_SOURCE} 
+	OPTIONS ${CUDA_DEVICE_CODE_FLAGS} -DCUDA) 
```

Comment:
I guess the reason why `cuda_add_library` is used here is that you used `find_package(CUDA)` instead of `enable_language(CUDA)`?